app/WebParsing/ReqParser.hs

```diff
@@ -425,6 +430,30 @@ courseAsModParser = do
     req <- andParser courseParser
     return $ REQUIREMENT req
 
+-- | Parses a literal "course" or "courses"
+courseLiteralParser :: Parser String
+courseLiteralParser = Parsec.spaces
+    >> Parsec.choice (map caseInsensitiveStr [
+        "courses",
+        "course"
+        ])
+
+-- | Parses a department code in the fces requirement
+-- | eg. the "CSC" in "1.0 credit in CSC" or "1.0 credit in CSC courses"
+departmentParser :: Parser Modifier
+departmentParser = do
+    Parsec.spaces
+    Parsec.notFollowedBy fceSeparator
+    dept <- many1Till Parsec.anyChar $ Parsec.choice (map Parsec.try [
+        courseLiteralParser,
+        Parsec.spaces >> Parsec.lookAhead fceSeparator >> return "",
```

Comment:
Same comment about spaces here as above. Also I'm not sure why `lookAhead` is needed here but not elsewhere.